autospec/specfiles.py

```diff
@@ -1583,17 +1623,30 @@ def write_ant_pattern(self):
         self._write_strip("")
 
     def write_gradle_pattern(self):
+        """Write gradle build pattern to spec file."""
         self.write_prep()
         self._write_strip("%build")
         self.write_build_prepend()
         self.write_proxy_exports()
-        self._write_strip("mkdir -p /builddir/.m2")
-        self._write_strip("cp -r /usr/share/java/.m2/* /builddir/.m2/")
-        self._write_strip("gradle --offline " + self.extra_make)
+        self._write_strip("mkdir -p %{buildroot}")
+        # It's ok if this doesn't exist
+        self._write_strip("cp -r /usr/share/java/.m2 ~/.m2 || :")
+        if self.subdir:
+            self._write_strip("pushd " + self.subdir)
+
+        # Point to our local maven repo, first
+        # style check does not like the escapes here
+        self._write_strip(r"sed -i 's|\(repositories\s*{\)|\1\n    mavenLocal()|' build.gradle")
+
+        self._write_strip("gradle --offline dependencies")
+        self._write_strip("gradle --offline build" + self.extra_make)
```

Comment:
Probably want an extra space after `build` in the string.

Reply:
Oops, yes.

Reply:
I'll also add `self.extra_make_install` in the appropriate place.